### DISCUSSION TITLE
Add a warning when we are trying to include a server-side module into the client code

### DISCUSSION THREAD

**Discussion by arunoda** [vercel/next.js#15999] [Ideas]
Recently, I imported `next-auth` into a client code and it started to throwing a lot of weird issues. Simple googling didn't help.
Then it seems like this I had to import `next-auth/client`.

It's not just me, [some others](https://github.com/nextauthjs/next-auth/issues/507#issuecomment-667322744) face this issue too.

So, it'll be nice if Next.js can render a warning if we try to import such a module. This will be a really nice touch to reduce human errors like this.

I'm not exactly sure on how to detect these modules, but something that depends on core NPM modules like `fs` & `http` is a good start.

**Comment by flybayer** [comment]
This is absolutely a very common issue in the field. I too think this should be built in. It definitely falls in line with the emphasis on helping people have small bundles. It’s way too easy to import a huge server library into frontend code.

**Comment by JosephChotard** [comment]
Would love an update on this feature! Very important imo
